### PR TITLE
Fixes VSTS Bug 810301: Accessibility: Version Control - Author:

### PR DESCRIPTION
**Pull request by mkrueger**
Authors section in the extreme right is not read by voice over

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/810301

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 1]
```diff
@@ -815,6 +842,50 @@ protected override bool OnExposeEvent (Gdk.EventExpose e)
...
+						accessibleSections.Add (new AuthorSectionAccessible (this, lineStart, ann, curY, nextY));
+						curY = nextY;
+					} else {
+						curY += widget.Editor.GetLineHeight (line);
+						line++;
+						widget.JumpOverFoldings (ref line);
+					}
+				}
+
+				Accessible.SetAccessibleChildren (accessibleSections.Select (a => a.Accessible).ToArray ());
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: This works, but the group has no annotation, VO just says "No annotation available". We should set the group role and a label.

Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: done.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			{
+				if (widget is null)
+					throw new ArgumentNullException (nameof (widget));
+				if (ann is null)
+					throw new ArgumentNullException (nameof (ann));
+
+				Accessible = AccessibilityElementProxy.ButtonElementProxy ();
+				Accessible.GtkParent = widget;
+
+				Accessible.SetRole (AtkCocoa.Roles.AXButton, GettextCatalog.GetString ("Blame annotation"));
```
[the remark this answers is not in the document]
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs]: done.

**Comment by sevoku** [comment]
@mkrueger the build failed, can you pls take a look?

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			{
+				if (widget is null)
+					throw new ArgumentNullException (nameof (widget));
+				if (ann is null)
+					throw new ArgumentNullException (nameof (ann));
+
+				Accessible = AccessibilityElementProxy.ButtonElementProxy ();
+				Accessible.GtkParent = widget;
+
+				Accessible.SetRole (AtkCocoa.Roles.AXMenuButton, GettextCatalog.GetString ("Blame annotation"));
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs]: Use `SetTitle` separately here, also let's call this "Authors Overview"

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs [hunk 4]
```diff
@@ -159,7 +160,9 @@ public BlameWidget (VersionControlDocumentInfo info)
 			editor.EditorOptionsChanged += delegate {
 				overview.OptionsChanged ();
 			};
-			editor.Caret.PositionChanged += ComparisonWidget.CaretPositionChanged;
+			editor.Caret.PositionChanged += HandleCaretPositionChanged;
+			overview.Accessible.Name = GettextCatalog.GetString ("Blame margin");
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: Let's call this "Authors Overview"

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs (hunk 3 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.AuthorSectionAccessible.cs]: +

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs (hunk 4 above)
Comment (mkrueger) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/BlameWidget.cs]: fixed.

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

**Pull request merged**